File: DiceGame/DiceGame_lua/Dice.lua
-- Meta class
Dice = {}
Dice.__index = Dice

--Constructor
function Dice:new()
    local o = {m_value = math.random(5)}
    setmetatable(o, self)
    return o
end

--Function
function Dice:roll()
    math.randomseed(tonumber(tostring({}):match("0x(.*)$"),16)) --LUA random generator sucks
    math.random() --First random number is not completely random, see comment above
    local number = math.random(5)
    self.m_value = number
end

--Function
function Dice:getValue()
    return self.m_value
end

File: MVC_O_DiceGame/DiceGame_java/model/Dice.java
package model;

public class Dice {
  
  private int m_value;
  
  public void roll() {
    m_value = (int)(Math.random() * 171717.0) % 6 + 1;
  }
  
  public int getValue() {
    return m_value;
  }
}